app/kubernetes/node/converter.js

```diff
@@ -54,6 +63,82 @@ class KubernetesNodeConverter {
     res.Yaml = yaml ? yaml.data : '';
     return res;
   }
+
+  static nodeToFormValues(node) {
+    const res = new KubernetesNodeFormValues();
+
+    res.Taints = _.map(node.Taints, (taint) => {
+      const res = new KubernetesNodeTaintFormValues();
+      res.Key = taint.Key;
+      res.Value = taint.Value;
+      res.Effect = taint.Effect;
+      res.NeedsDeletion = false;
+      res.IsNew = false;
+      return res;
+    });
+
+    res.Labels = _.map(node.Labels, (value, key) => {
+      const res = new KubernetesNodeLabelFormValues();
+      res.Key = key;
+      res.Value = value;
+      res.NeedsDeletion = false;
+      res.IsNew = false;
+      return res;
+    });
+
+    return res;
+  }
+
+  static formValuesToNode(node, formValues) {
```

Comment:
Only pass `formValues` and save the original node as `formValues.OriginalNode` before passing the `formValues`